plugins/source/azure/codegen/recipes/base.go

```diff
@@ -0,0 +1,260 @@
+package recipes
+
+import (
+	"fmt"
+	"log"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/codegen"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/gertd/go-pluralize"
+	"github.com/iancoleman/strcase"
+)
+
+type Template struct {
+	Source      string
+	Destination string
+}
+
+type Resource struct {
+	// Table is the table definition that will be used to generate the CloudQuery table
+	Table                    *codegen.TableDefinition
+	AzureStructName          string
+	AzurePackageName         string
+	AzureService             string
+	AzureSubService          string
+	Imports                  []string
+	Template                 Template
+	ListFunction             string
+	ListFunctionArgs         []string
+	ListFunctionArgsInit     []string
+	ListHandler              string
+	GetFunction              string
+	GetFunctionArgs          []string
+	GetFunctionArgsInit      []string
+	Helpers                  []string
+	MockHelpers              []string
+	MockListResult           string
+	MockListFunctionArgs     []string
+	MockListFunctionArgsInit []string
+	MockFieldsToIgnore       []string
+	MockValueType            string
+	MockDefinitionType       string
+}
+
+type template struct {
+	source            string
+	destinationSuffix string
+	imports           []string
+}
+
+type resourceDefinition struct {
```

Comment:
This is the resource configuration. Has some duplication with `Resource` but I thought it would be easier to separate the 2 as `Resource` is a transformed version of `resourceDefinition`